resources/type-system/strict-casts.md

```diff
@@ -0,0 +1,95 @@
+# Strict casts static analysis option
+
+This document specifies the "Strict casts" mode enabled with a static
+analysis option, new in Dart 2.16. As a static analysis option, we only intend
+to implement this feature in the Dart Analyzer. Under this feature, any
+implicit cast is reported as an analyzer "Hint" (a warning).
+
+Note that under the null safe type system, the only expressions which may be
+implicitly cast are those with a static type of `dynamic`. These are exactly
+the implicit casts reported in the "Strict casts" mode.
+
+## Enabling strict casts
+
+To enable strict casts, set the `strict-casts` option to `true`, under
+the Analyzer's `language` section:
+
+```yaml
+analyzer:
+  language:
+    strict-casts: true
+```
+
+## Motivation
+
+Static analysis can provide many benefits which require meaningful static
+types. For example, when a value of one type is _not assignable_ to a variable
+of another type, the error is reported by static analysis. However, when a
+value has a static type of `dynamic`, no error is reported; such a value is
```

Comment:
Grammar nit: I'd just use "." instead of ";". I've never found the semicolon to improve readability.

Reply:
I love semicolons, and very probably overuse them :). Done.